presentation/src/main/java/org/sagebionetworks/assessmentmodel/presentation/compose/ChoiceQuestionUi.kt

```diff
@@ -0,0 +1,204 @@
+        MaterialTheme.colors.primary//.copy(alpha = 0.12f)
+    } else {
+        MaterialTheme.colors.background
+    }
+    Surface(
+        shape = if (singleChoice) {
+            MaterialTheme.shapes.small.copy(CornerSize(100))
+        } else {
+            MaterialTheme.shapes.small},
+        elevation = Dp(5f),
+        modifier = Modifier
+            .padding(vertical = 8.dp)
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .background(answerBackgroundColor)
+                .clickable(
+                    onClick = {
+                        onChoiceSelected(!choiceSelected)
+                    }
+                )
+                .padding(vertical = 8.dp, horizontal = 8.dp),
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.Start
+        ) {
+            if (singleChoice) {
+                RadioButton(
+                    selected = choiceSelected,
+                    onClick = {
+                        onChoiceSelected(!choiceSelected)
+                    },
+                    colors = RadioButtonDefaults.colors(
+                        selectedColor = SageBlack
+                    )
+                )
+            } else {
+                Checkbox(
+                    checked = choiceSelected,
+                    onCheckedChange = { selected ->
+                        onChoiceSelected(selected)
+                    },
+                    colors = CheckboxDefaults.colors(
+                        checkedColor = SageBlack
+                    ),
+                )
+            }
+            when (inputItemState) {
+                is ChoiceInputItemStateImpl -> {
+                    Text(
+                        modifier = Modifier.padding(horizontal = 10.dp),
+                        text = inputItemState.inputItem.label,
+                        style = sageP1
+                    )
+                }
+
+                is KeyboardInputItemStateImpl<*> -> {
+                    val curAnswer = (inputItemState.currentAnswer as? JsonPrimitive)?.content ?: ""
+                    var text by remember { mutableStateOf(curAnswer) }
+                    Text(
+                        modifier = Modifier.padding(horizontal = 10.dp),
+                        text = inputItemState.inputItem.fieldLabel?: "Other:",
+                        style = sageP1
+                    )
+                    TextField(
+                        modifier = Modifier.padding(end = 20.dp),
+                        value = text,
+                        onValueChange = {
+                            text = it
+                            inputItemState.currentAnswer = JsonPrimitive(it)
+                            if (it.isNotEmpty()) {
+                                onChoiceSelected(true)
+                            }
+                        },
+                        singleLine = true
```

Comment:
design calls for wrapping the text where the "textfield" is really a multiple line text view (on iOS) with an inline prefix of "$fieldLabel: ". Super annoying to make that happen though so is this a WIP? Also, warning that once you get into supporting dynamic text sizes the "Other: " takes up a lot of real-estate.

Reply:
I Removed the singeLine. This is functional, but doesn't perfectly match the design. I'd like to walk through the interactions with Lynn at some point.